magefile.go

```diff
@@ -69,15 +71,43 @@ func Lint() error {
 		return err
 	}
 
-	if err := sh.RunV("go", "mod", "tidy"); err != nil {
-		return err
-	}
+	if err := filepath.WalkDir(".", func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
 
-	if err := sh.RunV("go", "work", "sync"); err != nil {
+		if d.IsDir() {
+			return nil
+		}
+
+		runGoModTidy := strings.HasSuffix(path, "go.mod")
+		if runGoModTidy {
+			cmd := exec.Command("go", "mod", "tidy")
+			cmd.Dir = filepath.Dir(path)
+			out, err := cmd.Output()
+			fmt.Printf(string(out))
+			if err != nil {
+				return err
+			}
+		}
+
+		runGoWorkSync := strings.HasSuffix(path, "go.work") || runGoModTidy
+		if runGoWorkSync {
+			cmd := exec.Command("go", "work", "sync")
```

Comment:
I don't think go work sync is supposed to be run in subdirectories (it probably does work, but it's meant to be a global command at top level). I have found that it seems to generally need to be run before go mod tidy too